basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/AASServerComponent.java

```diff
@@ -335,24 +337,29 @@ private List<ISubmodel> getSubmodelFromAggregator(IAASAggregator aggregator, IId
 
 		@SuppressWarnings("unchecked")
 		List<Object> submodelObject = (List<Object>) aasProvider.getValue(PREFIX_SUBMODEL_PATH);
-		
+
 		List<ISubmodel> persistentSubmodelList = new ArrayList<>();
-		
-		submodelObject.stream().map(this::getSubmodel).forEach(persistentSubmodelList::add);		
+
+		submodelObject.stream().map(this::getSubmodel).forEach(persistentSubmodelList::add);
 
 		return persistentSubmodelList;
 	}
-	
+
 	@SuppressWarnings("unchecked")
 	private ISubmodel getSubmodel(Object submodelObject) {
-		return Submodel.createAsFacade((Map<String, Object>) submodelObject);	
+		return Submodel.createAsFacade((Map<String, Object>) submodelObject);
 	}
 
 	private void loadAASServerFeaturesFromConfig() {
-		if(aasConfig.isPropertyDelegationEnabled()) {
+		if (aasConfig.isPropertyDelegationEnabled()) {
 			addAASServerFeature(new DelegationAASServerFeature());
 		}
 
+		String registryUrl = aasConfig.getRegistry();
+		if (!(registryUrl == null || registryUrl.isEmpty())) {
+			addAASServerFeature(new AutoRegisterAASServerFeature(registry, contextConfig.getUrl()));
```

Comment:
Please us getURL() instead of contextConfig.getUrl(). The latter may not be the right url in a cases of a containerized environment or when using a reverse proxy